common/cpp/verify_ias_report/ias_attestation_util.cpp

```diff
@@ -54,3 +64,30 @@ bool verify_quote(const std::string& ias_report, int group_out_of_date_is_ok) {
     return quote_status;
 }
 
+bool verify_mr_enclave_value(const std::string& enclave_quote_body,
+		             const std::string& mr_enclave) {
+    /* Verify MR enclave in the attestation
+     * report and compare with the value passed
+     * @param enclave_quote_body Enclave quote body
+     * @param mr_enclave MR enclave value in hex format
+     * Return true if comparision matches otherwise false
+     **/
+    if (mr_enclave.size() != 0) {
+        /* Extract ReportData and MR_ENCLAVE from isvEnclaveQuoteBody
+           present in Verification Report */
+	ByteArray quote_bytes = Base64EncodedStringToByteArray(
+	    enclave_quote_body.c_str());
+        sgx_quote_t* quote_body = reinterpret_cast<sgx_quote_t*>(
+            quote_bytes.data());
+        sgx_report_body_t* report_body = &quote_body->report_body;
+        sgx_measurement_t mr_enclave_from_report = *(&report_body->mr_enclave);
```

Comment:
The issue with above approach is it brings SGX SDK dependency on common/cpp module. So we need to install SGX SDK in client Dockerfile ? ALl this is done just to extract the mrencalve value from quote.

Reply:
I would suggest to do mrenclave verification at python layer either using 
1) Python struct - similar to 
    https://github.com/hyperledger/sawtooth-poet/blob/master/common/sawtooth_poet_common/sgx_structs/_sgx_quote.py
https://github.com/hyperledger/sawtooth-poet/blob/master/common/sawtooth_poet_common/sgx_structs/_sgx_report_body.py
https://github.com/hyperledger/sawtooth-poet/blob/master/common/tests/test_sgx_structs/tests.py

or

2) Extract the mrenclave value as bytes from a fixed offset in the IAS report

Reply:
Since we are using python client for doing to MR enclave check and we are depend on C++ sgx sdk. Is there problem if we bring SGX sdk dependency in common/cpp? Anyway we are using sgx sdk for verifying signature and quote verification. Porting C++ structure to python is not right thing to do. It may lead to inconsistencies. I am not clear on option 2.

Reply:
One of design goals of Avalon is to hide SGX complexities from Avalon clients. If we make Avalon clients install SGX SDK we are moving away from that goal.  Sawtooth PoET for example solves this by doing AVR verification at Python layer and it uses the python struct. Another option is used a fixed offset in AVR quote (based on IAS attestation API spec)
Also bringing in SGX SDK dependency on Avalon client just to do MREnclave verification is not necessary.